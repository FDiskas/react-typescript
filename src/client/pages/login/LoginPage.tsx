import * as React from 'react';
import * as styles from './LoginPage.scss';
import { NavLink } from 'react-router-dom';

export interface ILoginPageProps {}

export interface ILoginPageState {}

export class LoginPage extends React.Component<ILoginPageProps, ILoginPageState> {
    constructor(props: ILoginPageProps) {
        super(props);
    }

    render() {
        return (
            <main className={styles.page}>
                Component name: LoginPage
                <hr/>
                <NavLink to={'/'}>back to home</NavLink>
            </main>
        );
    }
}
