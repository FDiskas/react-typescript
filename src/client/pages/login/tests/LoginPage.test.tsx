import * as React from 'react';
import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

import { LoginPage } from '../LoginPage';

describe('LoginPage.js', () => {
    it('renders without crashing', () => {
        shallow(<LoginPage/>);
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<LoginPage/>);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });
});
