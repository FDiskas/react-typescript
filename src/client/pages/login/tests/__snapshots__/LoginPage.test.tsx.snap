// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage.js matches snapshot 1`] = `
<main
  className="page"
>
  Component name: LoginPage
  <hr />
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    to="/"
  >
    back to home
  </NavLink>
</main>
`;
